integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala

```diff
@@ -102,7 +115,7 @@ class CarbonScanRDD[V: ClassTag](
     val splits = carbonInputFormat.getSplits(job)
     if (!splits.isEmpty) {
       val carbonInputSplits = splits.asScala.map(_.asInstanceOf[CarbonInputSplit])
-
+      queryModel.setInvalidSegmentIds(validAndInvalidSegments.getInvalidSegments)
```

Comment:
move this to common getSplits, other wise validAndInvalidSegments can be null, if parallel deletion happens.